ar_read.c

```diff
@@ -113,12 +113,16 @@ static int ar_read(lua_State *L) {
     lua_getfield(L, 1, "reader"); // {ud}, {fenv}, fn
     if ( ! lua_isfunction(L, -1) ) err("MissingArgument: required parameter 'reader' must be a function");
     lua_setfield(L, -2, "reader"); // {ud}, {fenv}
-    lua_setfenv(L, -2); // {ud}
+#if LUA_VERSION_NUM > 501
```

Comment:
Will do. Note that I'm still working on this as the test suite crashes. There were some other changes in the Lua sandboxing that might be causing this. Still investigating.